packages/hub-extension/src/index.ts

```diff
@@ -78,8 +80,45 @@ function activateHubExtension(
         properties: {}
       }
     },
-    execute: () => {
-      window.open(restartUrl, '_blank');
+    execute: async () => {
+      // Create a widget for the iframe
+      const iframe = document.createElement('iframe');
+      iframe.src = restartUrl;
+      iframe.style.width = '70vw';
+      iframe.style.maxWidth = '100%';
+      iframe.style.height = '50vh';
+      iframe.style.maxHeight = '100%';
+      iframe.style.border = 'none';
+      // Wrap iframe in a Lumino Widget
+      const widget = new Widget({ node: iframe });
+
+      // Function to close the dialog
+      let dialog: Dialog<any> | null = null;
+      const onIframeNavigate = () => {
+        try {
+          if (iframe.contentWindow?.location) {
+            const currentUrl = new URL(iframe.contentWindow.location.href);
+            if (currentUrl.pathname.startsWith(labUrl)) {
+              dialog?.close();
+            }
+          } else {
+            throw Error('Could not get location of iframe');
+          }
+        } catch (e) {
+          console.warn('Error checking on iframe navigation:', e);
+          console.warn('Falling back to spawning in new window.');
+          window.open(restartUrl, '_blank');
+          dialog?.close();
+        }
+      };
+      iframe.addEventListener('load', onIframeNavigate);
+
+      dialog = new Dialog({
+        title: trans.__('Restarting Server...'),
```

Comment:
Let's use ellipsis character as it is used elsewhere

```suggestion
        title: trans.__('Restarting Server…'),
```